src/components/common/Filter/Filter.tsx

```diff
@@ -0,0 +1,177 @@
+import { useState } from 'react';
+import close from '@/assets/icons/close.svg';
+import closeRed from '@/assets/icons/close-red.svg';
+import { ADDRESS_OPTIONS } from '@/constants/dropdownOptions';
+import Button from '../Button';
+import Input from '../Input';
+import HorizontalLine from './HorizontalLine';
+
+type FilterProps = {
+  open: boolean;
+  onClose: () => void;
+  onApply: (values: {
+    address: string[];
+    startsAt: string | null;
+    hourlyPay: number | null;
+  }) => void;
+  defaultValues?: {
+    address?: string[];
+    startsAt?: string | null;
+    hourlyPay?: number | null;
+  };
+};
+
+export default function Filter({
+  open,
+  onClose,
+  onApply,
+  defaultValues,
+}: FilterProps) {
+  // 상태 정의
+  const [address, setAddress] = useState<string[]>(
+    defaultValues?.address ?? [],
+  );
+  const [startsAt, setStartsAt] = useState<string>(
+    defaultValues?.startsAt ?? '',
+  );
+  const [hourlyPay, setHourlyPay] = useState<string>(
+    defaultValues?.hourlyPay?.toString() ?? '',
+  );
+  const [inputType, setInputType] = useState<'text' | 'date'>('text');
+
+  // 주소 버튼 클릭 핸들러
+  const handleAddressClick = (addr: string) => {
+    setAddress((prev) =>
+      prev.includes(addr) ? prev.filter((a) => a !== addr) : [...prev, addr],
+    );
+  };
+
+  // 금액 입력 핸들러 (숫자만 허용, 3자리 콤마 적용)
+  const handlePayChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const rawValue = e.target.value.replace(/,/g, '').replace(/\D/g, '');
+    const formatted = rawValue.replace(/\B(?=(\d{3})+(?!\d))/g, ',');
+    setHourlyPay(formatted);
+  };
+
+  // 선택된 위치 삭제 핸들러
+  const handleDeleteLocation = (addr: string) => {
+    setAddress((prev) => prev.filter((item) => item !== addr));
+  };
+
+  // 초기화 버튼 핸들러
+  const handleReset = () => {
+    setAddress([]);
+    setStartsAt('');
+    setHourlyPay('');
+    setInputType('text');
+  };
+
+  // 적용하기 버튼 핸들러
+  const handleApply = () => {
+    const cleanPay = hourlyPay.replace(/,/g, '');
+    onApply({
+      address,
+      startsAt: startsAt || null,
+      hourlyPay: cleanPay ? Number(cleanPay) : null,
+    });
+    onClose();
+  };
+
+  if (!open) return null;
+
+  return (
+    <div className="z-[5] mx-auto flex w-full max-w-[390px] flex-col gap-[1.5rem] rounded-[1rem] border border-[#e5e4e7] bg-white p-[1.5rem] px-[1.25rem] shadow-[0_2px_8px_0_rgba(0,_0,_0,_0.25)]">
```

Comment:
```suggestion
    <div className="z-5 flex w-full max-w-390 flex-col gap-24 rounded-[10px] border border-gray-20 bg-white py-24 px-20 shadow-[0_2px_8px_0_rgba(0,_0,_0,_0.25)]">
```
💬 z-index는 그냥 숫자를 써도 됩니다! `mx-auto`는 필요하지 않은 것 같고요 아마 상위에서 absolute나 relative 등으로 위치를 지정할 것 같아 외부에서 className을 받거나 position 스타일을 지정해야 할 것 같습니다~ border-radius도 값이 맞지 않고 색깔 변수가 존재하면 변수를 이용해서 작성해주세요~ shadow도 커스텀 된 게 있으니 그걸 사용해 주세요~

Reply:
제가 아직 Css가 부족하다보니..많이 부족한데 많이 많이 알려주셔서 감사합니다!!

네 피드백 다 반영하고 수정해놓겠습니다!